File: s1237.js
var input = require("fs").readFileSync(__dirname + "/dev/stdin/input.txt", "utf8");
var lines = input.split("\r").join("").split("\n");

let a = lines.shift();
let b = lines.shift();
while (a && b) {
  let maior = 0;
  for (let u = 0; u < a.length; u++) {
    for (let v = b.length - 1; v >= 0; v--) {
      let i = u;
      let j = v;
      let iguais = 0;
      let maiorInterno = 0;
      while (i < a.length && j < b.length) {
        if (a[i] == b[j]) {
          iguais++;
        } else {
          iguais = 0;
        }
        if (maiorInterno < iguais) {
          maiorInterno = iguais;
        }
        i++;
        j++;
      }
      if (maior < maiorInterno) {
        maior = maiorInterno;
      }
    }
  }
  console.log(maior);
  a = lines.shift();
  b = lines.shift();
}

File: s1237.c
#include <stdio.h>

int main(void)
{
  int maior, u, v, i, j, iguais, maiorInterno;
  char a[200], b[200];
  while (scanf(" %[^\n]s", a) > 0 && scanf(" %[^\n]s", b) > 0)
  {
    maior = 0;
    for (u = 0; a[u] != '\0'; u++)
    {
      for (v = 0; b[v] != '\0'; v++)
        ;
      for (v = v - 1; v >= 0; v--)
      {
        i = u;
        j = v;
        iguais = 0;
        maiorInterno = 0;
        while (a[i] != '\0' && b[j] != '\0')
        {
          if (a[i] == b[j])
          {
            iguais++;
          }
          else
          {
            iguais = 0;
          }
          if (maiorInterno < iguais)
          {
            maiorInterno = iguais;
          }
          i++;
          j++;
        }
        if (maior < maiorInterno)
        {
          maior = maiorInterno;
        }
      }
    }
    printf("%d\n", maior);
  }

  return 0;
}